apps/newsletters-api/src/app/authorisation.ts

```diff
@@ -0,0 +1,42 @@
+import type { FastifyRequest } from 'fastify/types/request';
+import type { UserProfile } from '@newsletters-nx/newsletters-data-client';
+import {
+	getUserEditSchema,
+	isPartialNewsletterData,
+} from '@newsletters-nx/newsletters-data-client';
+import { permissionService } from '../services/permissions';
+
+export const hasEditAccess = async (
+	profile: UserProfile | undefined,
+): Promise<boolean> => {
+	if (!profile) return false;
+	const permissions = await permissionService.get(profile);
+	const { editTags, editNewsletters, editSignUpPage, editBraze, editOphan } =
+		permissions;
+	return [editTags, editNewsletters, editSignUpPage, editBraze, editOphan].some(
+		(permission) => permission,
+	);
+};
+
+export const isAuthorisedToUpdateNewsletter = async (
+	profile: UserProfile | undefined,
+	request: FastifyRequest,
```

Comment:
I would tend to extract the body (and maybe parse it using isPartialNewsletter) from the request and use that as the argument rather than sending the whole request to this function.

That way we only need to think about the API using Fastify within the modules that deal with the routes - it would also make the function easier to test